### PR TITLE
cunicu: init at 0.5.53

### PR DESCRIPTION
**Pull request by stv0g**
[cunīcu](https://github.com/cunicu/cunicu) is a user-space daemon managing [WireGuard®](https://wireguard.com/) interfaces to establish a mesh of peer-to-peer VPN connections in harsh network environments.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cu/cunicu/package.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  postInstall = ''
+    install -d $out/usr/share/man/man1
+    install ./docs/usage/man/*.1 $out/usr/share/man/man1
+    install -D <(cunicu completion bash) $out/share/bash-completion/completions/cunicu
+    install -D <(cunicu completion fish) $out/share/fish/vendor_completions.d/cunicu.fish
+    install -D <(cunicu completion zsh) $out/share/zsh/vendor-completions/_cunicu
+  '';
+
+  meta = {
+    description = "A zeroconf peer-to-peer mesh VPN using Wireguard® and Interactive Connectivity Establishment (ICE)";
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:63]:
```suggestion
    description = "Zeroconf peer-to-peer mesh VPN using Wireguard® and Interactive Connectivity Establishment (ICE)";
```

File: pkgs/by-name/cu/cunicu/package.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    homepage = "https://cunicu.li";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ stv0g ];
+    platforms = [
+      "aarch64-linux"
+      "x86_64-linux"
+    ];
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:70]: missing `mainProgram`?

File: pkgs/by-name/cu/cunicu/package.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    "pkg/selfupdate"
+    "pkg/tty"
+    "scripts"
+  ];
+
+  ldflags = [
+    "-X"
+    "cunicu.li/cunicu/pkg/buildinfo.Version=${version}"
+    "-X"
+    "cunicu.li/cunicu/pkg/buildinfo.BuiltBy=Nix"
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:42]:
```suggestion
    "-X cunicu.li/cunicu/pkg/buildinfo.Version=${version}"
    "-X cunicu.li/cunicu/pkg/buildinfo.BuiltBy=Nix"
```

File: pkgs/by-name/cu/cunicu/package.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  ];
+
+  preBuild = ''
+    go generate ./...
+  '';
+
+  postBuild = ''
+    PATH=$PATH:$GOPATH/bin
+    cunicu docs --with-frontmatter
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:52]:
```suggestion
```

File: pkgs/by-name/cu/cunicu/package.nix [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  postInstall = ''
+    install -d $out/usr/share/man/man1
+    install ./docs/usage/man/*.1 $out/usr/share/man/man1
+    install -D <(cunicu completion bash) $out/share/bash-completion/completions/cunicu
+    install -D <(cunicu completion fish) $out/share/fish/vendor_completions.d/cunicu.fish
+    install -D <(cunicu completion zsh) $out/share/zsh/vendor-completions/_cunicu
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:60]:
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    cunicu docs --with-frontmatter
    installManpage ./docs/usage/man/*.1
    installShellCompletion \
      --bash <(cunicu completion bash) \
      --zsh <(cunicu completion zsh) \
      --fish <(cunicu completion fish)
  '';
```

https://nixos.org/manual/nixpkgs/unstable/#installshellfiles

File: pkgs/by-name/cu/cunicu/package.nix [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    owner = "cunicu";
+    repo = "cunicu";
+    rev = "v${version}";
+    hash = "sha256-y8JyXf+LdquhVr9G5wX1LdMF7wpj5PZG3xXGAWoypzA=";
+  };
+
+  nativeBuildInputs = [
+    protobuf
+    protoc-gen-go
+    protoc-gen-go-grpc
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:23]:
```suggestion
    protoc-gen-go-grpc
    installShellFiles
```

File: pkgs/by-name/cu/cunicu/package.nix (hunk 2 above)
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:70]:
```suggestion
    platforms = lib.platforms.linux;
```

File: pkgs/by-name/cu/cunicu/package.nix [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  ];
+
+  CGO_ENABLED = 0;
+
+  vendorHash = "sha256-OiLVdEf6fcGHx0k0xC5sZwhnK0FiLgfdkz2zNgBbcgY=";
+
+  # These packages contain networking dependent tests which fail in the sandbox
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:30]: But are they needed for the functionality?

File: pkgs/by-name/cu/cunicu/package.nix (hunk 2 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Fixed in 295d96d041d31a0301f5f7b5599a232e5a537f3c

File: pkgs/by-name/cu/cunicu/package.nix (hunk 6 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix:26]: Fixed in 295d96d041d31a0301f5f7b5599a232e5a537f3c

File: pkgs/by-name/cu/cunicu/package.nix (hunk 5 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Fixed in 295d96d041d31a0301f5f7b5599a232e5a537f3c

File: pkgs/by-name/cu/cunicu/package.nix (hunk 4 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Fixed in 295d96d041d31a0301f5f7b5599a232e5a537f3c

File: pkgs/by-name/cu/cunicu/package.nix (hunk 2 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Fixed in 295d96d041d31a0301f5f7b5599a232e5a537f3c

File: pkgs/by-name/cu/cunicu/package.nix (hunk 1 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Fixed in 295d96d041d31a0301f5f7b5599a232e5a537f3c

File: pkgs/by-name/cu/cunicu/package.nix (hunk 3 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix:45]: This snipped has been formatted like this by `nixfmt`.

File: pkgs/by-name/cu/cunicu/package.nix (hunk 7 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix:33]: No the tests are not crucial for the functionality.
I am working on a NixOS test and corresponding NixOS module for cunicu (I am the upstream maintainer as well).

File: pkgs/by-name/cu/cunicu/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/cu/cunicu/package.nix:45]: This is not about formatting

File: pkgs/by-name/cu/cunicu/package.nix [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    "-X"
+    "cunicu.li/cunicu/pkg/buildinfo.BuiltBy=Nix"
+  ];
+
+  preBuild = ''
+    go generate ./...
+  '';
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    cunicu docs --with-frontmatter
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:53]:
```suggestion
    $out/bin/cunicu docs --with-frontmatter
```

See https://github.com/NixOS/nixpkgs/pull/354164/checks?check_run_id=32744992651

Comment (Aleksanaa) [reply pkgs/by-name/cu/cunicu/package.nix:53]: Also below

Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Thanks, fixed in 5833f77de019

File: pkgs/by-name/cu/cunicu/package.nix (hunk 3 above)
Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Oh, I missed that.

Fixed in f8e76a836715

File: pkgs/by-name/cu/cunicu/package.nix [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    "-X cunicu.li/cunicu/pkg/buildinfo.BuiltBy=Nix"
+  ];
+
+  preBuild = ''
+    go generate ./...
+  '';
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    $out/bin/cunicu docs --with-frontmatter
+    installManpage ./docs/usage/man/*.1
```
Comment (Aleksanaa) [inline pkgs/by-name/cu/cunicu/package.nix:52]:
```suggestion
    installManPage ./docs/usage/man/*.1
```

Comment (Aleksanaa) [reply pkgs/by-name/cu/cunicu/package.nix:52]: Please test locally first

Comment (stv0g) [reply pkgs/by-name/cu/cunicu/package.nix]: Sorry, fixed and built in aarch64-linux in 1b6ddea547bb1d215c855de4a0ce0ced93803500

**Pull request merged**